pkg/daemon/ceph/osd/kms/k8s.go

```diff
@@ -54,6 +54,24 @@ func (c *Config) storeSecretInKubernetes(pvcName, key string) error {
 	return nil
 }
 
+// updateSecretInKubernetes updates the dmcrypt key in a Kubernetes Secret
+func (c *Config) updateSecretInKubernetes(pvcName, key string) error {
+	secretName := GenerateOSDEncryptionSecretName(pvcName)
+	secret, err := c.context.Clientset.CoreV1().Secrets(c.ClusterInfo.Namespace).Get(c.ClusterInfo.Context, secretName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get secret %q", secretName)
+	}
+
+	secret.StringData = map[string]string{OsdEncryptionSecretNameKeyName: key}
+	// Update the Kubernetes Secret
+	_, err = c.context.Clientset.CoreV1().Secrets(c.ClusterInfo.Namespace).Update(c.ClusterInfo.Context, secret, metav1.UpdateOptions{})
+	if err != nil && !kerrors.IsAlreadyExists(err) {
```

Comment:
Does `Update()` ever return an error that would result in `already exists`? I thought that would just be after calling `Create()`

Reply:
yes, its not required. I must have missed removing it from my earlier trials, removed it now.
thanks.